src/application/types/Wallet.ts

```diff
@@ -0,0 +1,158 @@
+import {
+    Coin,
+  getCoinId,
+  masterPublicKeyToFirstPuzzleHash,
+  masterPublicKeyToWalletSyntheticKey,
+  masterSecretKeyToWalletSyntheticSecretKey,
+  Peer,
+  puzzleHashToAddress,
+  secretKeyToPublicKey,
+  selectCoins,
+  signMessage,
+} from '@dignetwork/datalayer-driver';
+import { mnemonicToSeedSync } from 'bip39';
+import { PrivateKey } from 'chia-bls';
+import { FileCacheService } from '../services/FileCacheService';
+import { CACHE_DURATION } from '../services/WalletService';
+import path from 'path';
+import os from 'os';
+
+export const MIN_HEIGHT = 5777842;
+export const MIN_HEIGHT_HEADER_HASH =
+  "b29a4daac2434fd17a36e15ba1aac5d65012d4a66f99bed0bf2b5342e92e562c";
+  
+export const USER_DIR_PATH = path.join(os.homedir(), '.dig');
+
+export class Wallet {
+  private mnemonic: string;
+
+  public constructor(mnemonic: string) {
+    this.mnemonic = mnemonic;
+  }
+
+  public getMnemonic(): string {
+    if (!this.mnemonic) {
+      throw new Error('Mnemonic seed phrase is not loaded.');
+    }
+    return this.mnemonic;
+  }
+
+  public async getMasterSecretKey(): Promise<Buffer> {
+    const seed = mnemonicToSeedSync(this.getMnemonic());
+    return Buffer.from(PrivateKey.fromSeed(seed).toHex(), 'hex');
+  }
+
+  public async getPublicSyntheticKey(): Promise<Buffer> {
+    const master_sk = await this.getMasterSecretKey();
+    const master_pk = secretKeyToPublicKey(master_sk);
+    return masterPublicKeyToWalletSyntheticKey(master_pk);
+  }
+
+  public async getPrivateSyntheticKey(): Promise<Buffer> {
+    const master_sk = await this.getMasterSecretKey();
+    return masterSecretKeyToWalletSyntheticSecretKey(master_sk);
+  }
+
+  public async getOwnerPuzzleHash(): Promise<Buffer> {
+    const master_sk = await this.getMasterSecretKey();
+    const master_pk = secretKeyToPublicKey(master_sk);
+    return masterPublicKeyToFirstPuzzleHash(master_pk);
+  }
+
+  public async getOwnerPublicKey(): Promise<string> {
+    const ownerPuzzleHash = await this.getOwnerPuzzleHash();
+    return puzzleHashToAddress(ownerPuzzleHash, 'xch');
+  }
+
+  public async createKeyOwnershipSignature(nonce: string): Promise<string> {
+    const message = `Signing this message to prove ownership of key.\n\nNonce: ${nonce}`;
+    const privateSyntheticKey = await this.getPrivateSyntheticKey();
+    const signature = signMessage(Buffer.from(message, 'utf-8'), privateSyntheticKey);
+    return signature.toString('hex');
+  }
+
+  public async selectUnspentCoins(
+    peer: Peer,
+    coinAmount: bigint,
+    feeBigInt: bigint,
+    omitCoins: Coin[] = []
+  ): Promise<Coin[]> {
+    const cache = new FileCacheService<{ coinId: string; expiry: number }>(
```

Comment:
Just a heads up what reserved coins are.

I ran into a situation where I would spend a coin, and it was sent to the memepool. But it was still considered unspent until the transaction confirmed.

Reserve coins are a registry of coins I already used in prior operations. This prevented a race condition to prevent accidentally selecting coins to spend that were already sent to the memepool.